arangod/Agency/Agent.cpp

```diff
@@ -253,11 +253,11 @@ AgentInterface::raft_commit_t Agent::waitFor(index_t index, double timeout) {
     ///  (_waitForCV's mutex stops writes to _commitIndex)
     CONDITION_LOCKER(guard, _waitForCV);
     if (leading()) {
-      if (lastCommitIndex != _commitIndex) {
+      if (lastCommitIndex != _commitIndex.load(std::memory_order_relaxed)) {
```

Comment:
Here we are loading `_commitIndex` for a comparison, and 3 or 4 lines later we are loading it again unconditionally. We could fuse the two loads into just one.